src/main/java/xyz/nucleoid/plasmid/game/stats/GameStatisticBundle.java

```diff
@@ -0,0 +1,81 @@
+package xyz.nucleoid.plasmid.game.stats;
+
+import com.google.gson.JsonObject;
+import it.unimi.dsi.fastutil.objects.Object2ObjectMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.minecraft.server.network.ServerPlayerEntity;
+import xyz.nucleoid.plasmid.util.PlayerRef;
+
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Wrapper containing a map players' {@link StatisticBundle}s
+ */
+public class GameStatisticBundle {
+    private final Object2ObjectMap<UUID, StatisticBundle> players = new Object2ObjectOpenHashMap<>();
+
+    public StatisticBundle getPlayer(PlayerRef player) {
+        return this.getPlayer(player.getId());
+    }
+
+    public StatisticBundle getPlayer(ServerPlayerEntity player) {
+        return this.getPlayer(player.getUuid());
+    }
+
+    public StatisticBundle getPlayer(UUID uuid) {
+        if (!this.players.containsKey(uuid)) {
+            this.players.put(uuid, new StatisticBundle());
+        }
+        return this.players.getOrDefault(uuid, new StatisticBundle());
+    }
+
+    // Gotta love generics
+    public void increment(ServerPlayerEntity player, StatisticKey<Double> key, double amount) {
```

Comment:
It's probably fine to keep these functions just on the `StatisticsBundle`, as `getPlayer(..).increment(..)` seems more clear in any case.

Also means a lot less duplication for each type of player parameter. 😄